src/nodedef.cpp

```diff
@@ -682,6 +682,8 @@ void ContentFeatures::updateTextures(ITextureSource *tsrc, IShaderSource *shdsrc
 	switch (drawtype) {
 	default:
 	case NDT_NORMAL:
+		material_type = (alpha == 255) ?
```

Comment:
I agree, however, that should not get in the way of what is essentially a bugfix. This PR, as it stands, should not break any mod other than those abusing the z-sorting issue when transparent textures are applied to normal nodes.